### PR TITLE
ref(ts) Convert activity inputs to typescript

### PR DESCRIPTION
**Pull request by markstory**
I needed to juggle type definitions for a few of the alert types and components as they were doing some unconventional things with prop spreading.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/activity/note/input.tsx [hunk 1]
```diff
@@ -12,83 +12,66 @@ import marked from 'app/utils/marked';
...
-  cleanMarkdown(text) {
+  cleanMarkdown(text: string) {
     return text
       .replace(/\[sentry\.strip:member\]/g, '@')
       .replace(/\[sentry\.strip:team\]/g, '');
   }
 
-  submitForm = () => {
+  submitForm() {
     if (!!this.props.modelId) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/activity/note/input.tsx:67]: as you are touching this file, maybe you could use an early return here avoiding the usage of else 🙃

File: src/sentry/static/sentry/app/components/activity/note/input.tsx [hunk 2]
```diff
@@ -131,40 +114,42 @@ class NoteInput extends React.Component {
...
     this.setState({value: e.target.value});
 
     if (this.props.onChange) {
       this.props.onChange(e, {updating: !!this.props.modelId});
     }
   };
 
-  handleKeyDown = e => {
+  handleKeyDown = (
+    e: React.KeyboardEvent<HTMLTextAreaElement> | React.KeyboardEvent<HTMLInputElement>
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/activity/note/input.tsx:131]:
```suggestion
    e: React.KeyboardEvent<HTMLTextAreaElement | HTMLInputElement>
```

File: src/sentry/static/sentry/app/components/activity/note/input.tsx [hunk 3]
```diff
@@ -131,40 +114,42 @@ class NoteInput extends React.Component {
...
   };
 
-  handleCancel = e => {
+  handleCancel = (e: React.MouseEvent<Element>) => {
     e.preventDefault();
     this.finish();
   };
 
-  handleAddMember = (id, display) => {
+  handleAddMember = (id: React.ReactText, display: string) => {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/activity/note/input.tsx:144]: I've seen that we usually use ReactText (number | string) but then we convert to string. shouldn't we only use the type string instead?  :D

File: src/sentry/static/sentry/app/components/activity/note/types.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * List of id, display name
+ */
+export type Mentioned = [string, string];
+
+/**
+ * The typings for the react-mentionables library use this type
+ * for their change event so we also use it.
+ */
+export type MentionChangeEvent = {target: {value: string}};
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/activity/note/types.tsx:19]: should we use maybe:
`import {OnChangeHandlerFunc} from 'react-mentions';`
`export type MentionChangeEvent = Parameters<OnChangeHandlerFunc>[0]` ?

File: src/sentry/static/sentry/app/components/activity/note/index.tsx [hunk 5]
```diff
@@ -1,52 +1,68 @@
...
+import ActivityItem, {ActivityAuthorType} from 'app/components/activity/item';
 import space from 'app/styles/space';
 
 import EditorTools from './editorTools';
 import NoteBody from './body';
 import NoteHeader from './header';
 import NoteInput from './input';
 
-class Note extends React.Component {
-  static propTypes = {
```
[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/static/sentry/app/components/activity/note/index.tsx:15]: All of the places that use this component are in typescript from what I could find so proptypes aren't needed anymore.

File: src/sentry/static/sentry/app/components/activity/note/input.tsx [hunk 6]
```diff
@@ -12,83 +12,66 @@ import marked from 'app/utils/marked';
...
       .replace(/\[sentry\.strip:team\]/g, '');
   }
 
-  submitForm = () => {
+  submitForm() {
     if (!!this.props.modelId) {
       this.update();
-    } else {
-      this.create();
+      return;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/activity/note/input.tsx:69]: 💘

File: src/sentry/static/sentry/app/components/activity/note/input.tsx [hunk 7]
```diff
@@ -269,12 +252,18 @@ class NoteInput extends React.Component {
...
-  };
+type NoteInputContainerProps = {
+  projectSlugs: string[];
+} & Omit<Props, 'memberList' | 'teams'>;
+
+type MentionablesChildFunc = Parameters<
+  React.ComponentProps<typeof Mentionables>['children']
+>[0];
+
+class NoteInputContainer extends React.Component<NoteInputContainerProps> {
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/activity/note/input.tsx:261]: sounds good...thank you!

File: src/sentry/static/sentry/app/components/activity/note/types.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * List of id, display name
+ */
+export type Mentioned = [string, string];
+
+/**
+ * The typings for the react-mentionables library use this type
+ * for their change event so we also use it.
+ */
+export type MentionChangeEvent = Parameters<OnChangeHandlerFunc>[0];
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/activity/note/types.tsx:21]: 👏

**Review by priscilawebdev — APPROVED** [review]
✅✅✅✅✅

**Pull request merged**